modules/data/src/dataservices/interfaces.ts

```diff
@@ -24,9 +31,28 @@ export interface RequestData {
 
 /**
  * A key/value map of parameters to be turned into an HTTP query string
- * Same as HttpClient's HttpParamsOptions which is NOT exported at package level
+ * Same as HttpClient's HttpParamsOptions which at the time of writing was
+ * NOT exported at package level
  * https://github.com/angular/angular/issues/22013
+ *
+ * @deprecated Use HttpOptions instead. getWithQuery still accepts QueryParams as its
+ * first argument, but HttpOptions.httpParams uses Angular's own HttpParamsOptions which
+ * HttpClient accepts as an argument.
  */
 export interface QueryParams {
-  [name: string]: string | string[];
+  [name: string]:
+    | string
+    | number
+    | boolean
+    | ReadonlyArray<string | number | boolean>;
+}
+
+/**
+ * Options that adhere to the constructor arguments for HttpParams and
+ * HttpHeaders.
+ */
+export interface HttpOptions {
+  httpParams?: HttpParamsOptions;
+  httpHeaders?: string | { [p: string]: string | string[] };
+  [key: string]: any;
```

Comment:
Why do we need to use the index property here?

Reply:
Wanted to leave the type open so user's could pass custom properties through. Had angular's newish context concept in mind, but happy to remove it if preferred.

Reply:
[HttpContext](https://angular.io/api/common/http/HttpContext)

I'd thought users overriding the default data service may want to be able to pass arbitrary data via the action so they can set HttpContext properties.

Happy to remove it though, I don't need it myself, just seemed potentially useful.

Reply:
If you don't need it, I would say to leave it out.
Otherwise we broaden our surface without anyone using it.
If it's needed in the future we can always add it.

Reply:
done